RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ChromeCacheExtractor.java

```diff
@@ -0,0 +1,1363 @@
+                dataSizes[i] = fileROBuf.getInt();
+            }
+            dataAddresses = new CacheAddress[4];
+            for (int i = 0; i < 4; i++)  {
+                dataAddresses[i] =  new CacheAddress(fileROBuf.getInt() & UINT32_MASK, selfAddress.getCachePath());
+            }
+        
+            flags = fileROBuf.getInt() & UINT32_MASK;
+            // skip over pad 
+            for (int i = 0; i < 4; i++)  {
+                fileROBuf.getInt();
+            }
+            
+            // skip over self hash
+            fileROBuf.getInt();
+        
+            // get the key
+            if (longKeyAddresses != null) {
+                // Key is stored outside of the entry
+                try {
+                    CacheData data = new CacheData(longKeyAddresses, this.keyLen, true);
+                    key = data.getDataString();
+                } catch (TskCoreException | IngestModuleException ex) {
+                    logger.log(Level.SEVERE, String.format("Failed to get external key from address %s", longKeyAddresses)); //NON-NLS 
+                } 
+            }
+            else {  // key stored within entry 
+                StringBuilder strBuilder = new StringBuilder(MAX_KEY_LEN);
+                int keyLen = 0;
+                while (fileROBuf.remaining() > 0  && keyLen < MAX_KEY_LEN)  {
+                    char keyChar = (char)fileROBuf.get();
+                    if (keyChar == '\0') { 
+                        break;
+                    }
+                    strBuilder.append(keyChar);
+                    keyLen++;
+                }
+
+                key = strBuilder.toString();
+            }
+        }
+
+        public CacheAddress getAddress() {
+            return selfAddress;
+        }
+
+        public long getHash() {
+            return hash;
+        }
+
+        public CacheAddress getNextAddress() {
+            return nextAddress;
+        }
+
+        public int getReuseCount() {
+            return reuseCount;
+        }
+
+        public int getRefetchCount() {
+            return refetchCount;
+        }
+
+        public EntryStateEnum getState() {
+            return state;
+        }
+
+        public long getCreationTime() {
+            return creationTime;
+        }
+
+        public long getFlags() {
+            return flags;
+        }
+
+        public String getKey() {
+            return key;
+        }
+        
+        /**
+         * Returns the data segments in the cache entry.
+         * 
+         * @return list of data segments in the entry.
+         * 
+         * @throws TskCoreException
+         * @throws org.sleuthkit.autopsy.ingest.IngestModule.IngestModuleException 
+         */
+        public List<CacheData> getData() throws TskCoreException, IngestModuleException {
+            
+            if (dataList == null) { 
+                dataList = new ArrayList<>();
+                 for (int i = 0; i < 4; i++)  {
```

Comment:
I suggest using a multi-character var instead of i and putting the 4 into a constant.